include/debug/stack.h

```diff
@@ -82,11 +83,28 @@ static inline void stack_analyze(const char *name, const char *stack,
  * @param name Name of the stack
  * @param sym The symbol of the stack
  */
-#define STACK_ANALYZE(name, sym)				\
+#define STACK_ANALYZE(name, sym) DEPRECATED_MACRO		\
 	do {							\
 		stack_analyze(name,				\
 			      Z_THREAD_STACK_BUFFER(sym),	\
 			      K_THREAD_STACK_SIZEOF(sym));	\
 	} while (false)
 
+static inline void log_stack_usage(const char *name,
+				   const struct k_thread *thread)
+{
+#if defined(CONFIG_INIT_STACKS) && defined(CONFIG_THREAD_STACK_INFO)
+	size_t unused, size = thread->stack_info.size;
+
+	LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
+
+	if (k_thread_stack_space_get(thread, &unused) == 0) {
+		unsigned int pcnt = ((size - unused) * 100U) / size;
+
+		LOG_INF("%p (%s):\tunused %zu\tusage %zu / %zu (%u %%)",
```

Comment:
How about we get the thread name using k_thread_name_get() if name is NULL, or we could just set the thread name for those threads and remove the argument completely. The macro predates the support for setting thread names...

Reply:
do you want this done here or can it be in a separate PR?
I hesitate to make this function depend on 3 separate kconfigs but perhaps we must

Reply:
You do not need to depend on CONFIG_THREAD_NAME here afaik, you can just do:
```
tname = k_thread_name_get(thread);
if (tname != NULL) {
   ...
}
```
Having free-style thread names here as in the old macro is not great, we should just set the thread name in one place, so it matches everything else, afaik thread names are now also being used in exceptions and other places. And since this is a debug feature, the extra overhead is acceptable.

Reply:
@nashif this is now done